### ISSUE TITLE
Slim weights vs Full weights & Finetuning

### ISSUE THREAD

**Issue by msakthiganesh** [kingoflolz/mesh-transformer-jax#98]
Hi,

I am able to perform inferencing with slim weights and the results are pretty good. I was wondering if we can finetune the GPT-J slim weights model as mentioned in the link:  [https://github.com/mallorbc/Finetune_GPTNEO_GPTJ6B](https://github.com/mallorbc/Finetune_GPTNEO_GPTJ6B) by converting the model to PyTorch format.

My understanding is that:
**Slim Weights:** contains only the model weights, can be used only for inferencing, doesn't contain optimizer states and hence can't be fine-tuned.
**Full Weights:** Includes the optimizer params and hence can be fine-tuned for downstream tasks.

Can you please confirm if finetuning with slim weights is actually possible?

Thanks!

**Comment by kingoflolz** [comment]
Fine tuning with slim weights is possible, however you will likely need a longer warmup period to avoid destroying the existing trained structure.

**Comment by niyoushanajmaei** [comment]
@msakthiganesh 
Did you end up fine tuning the slim weights with a longer warmup period successfully? 
If yes, what percentage of the total steps did you use for warmup?

**Comment by msakthiganesh** [comment]
Hi @niyoushanajmaei. No, I did not try fine-tuning the slim weights model. But I guess, warmup steps around 4k should work. Would be better if someone from the team can comment.
